lib/api/controllers/remoteActions/managePlugins.js

```diff
@@ -0,0 +1,591 @@
+  });
+
+  return q.all(promises);
+}
+
+
+/**
+ * Detects if the configured plugin must be installed
+ * If the plugin is configured with an url from GIT, the plugin is installed every time
+ *   to ensure getting the latest release
+ * If the plugin come from NPM or , the plugin is installed only if the required version
+ *   is different from the version of the already installed plugin
+ *
+ * @param plugin
+ * @param name
+ * @returns {boolean} true if the plugin must be installed, false if not
+ */
+function needInstall(plugin, name) {
+  var
+    packageDefinition,
+    packagePath,
+    lockPath,
+    pluginPath = getPathPlugin(plugin, name);
+
+  // If we want to install a plugin with git, maybe there is no version and we want to 'pull' the plugin
+  if (plugin.gitUrl) {
+    lockPath = path.join(pluginPath, 'lock');
+    // Check if the plugin was installed in the current hour, if not then we install it again in case of update
+    if (fs.existsSync(lockPath)) {
+      if (new Date(fs.statSync(lockPath).mtime) > new Date() - 3600000 === true) {
+        return false;
+      }
+      fs.unlinkSync(lockPath);
+    }
+    return true;
+  }
+
+  if (plugin.path) {
+    return false;
+  }
+
+  packagePath = path.join(pluginPath, 'package.json');
+  if (!fs.existsSync(packagePath)) {
+    return true;
+  }
+
+  packageDefinition = require(path.join(pluginPath, 'package.json'));
+
+  // If version in package.json is different from the version the plugins.json, we want to install the updated plugin
+  return (packageDefinition._from !== name + '@' + plugin.npmVersion);
+}
+
+/**
+ * Return the real plugin path
+ * @param plugin
+ * @param name
+ * @returns {String}
+ */
+function getPathPlugin (plugin, name) {
+  if (plugin.path) {
+    return plugin.path;
+  }
+  return path.join(__dirname, '..', '..', '..', '..', 'node_modules', name);
+}
+
+/**
+ * Dumps a plugin's configuration on the standard output.
+ *
+ * @param {string} plugin name
+ * @param {object} db - kuzzle database handler
+ * @param {object} cfg - kuzzle configuration
+ * @returns {Promise}
+ */
+function getPluginConfiguration(plugin, db, cfg) {
+  return db
+    .get(cfg.pluginsManager.dataCollection, plugin)
+    .then(result => {
+      console.dir(result._source, {depth: null});
+      return result._source;
+    });
+}
+
+/**
+ * Sets a new plugin property
+ *
+ * @param {string} plugin
+ * @param {string} property - expected: stringified JSON object
+ * @param {object} db - kuzzle database handler
+ * @param {object} cfg - kuzzle configuration
+ * @returns {Promise}
+ */
+function setPluginConfiguration(plugin, property, db, cfg) {
+  var jsonProperty;
+
+  try {
+    jsonProperty = JSON.parse(property);
+  }
+  catch (err) {
+    return q.reject(new Error('Unable to parse ' + property + '. Expected: JSON Object\n' + err));
+  }
+
+  return db
+    .get(cfg.pluginsManager.dataCollection, plugin)
```

Comment:
Well if update merges the actual document with the new changes, then no we don't have to do it manually with _.extend